packages/minter-contracts/src/michelson-to-ts.ts

```diff
@@ -66,43 +65,67 @@ const compile = (contracts: string[]) =>
     // This does incur an additional decode->encode cycle in the script, but
     // the resulting file size savings is worth the additional roundtrip.
     const json = JSON.parse(await michelsonToJson(path.join(michelsonPath, fileName)));
-    const alias = fileNameToAlias(fileName);
     await fs.promises.writeFile(
-      path.join(outPath, `${fileName}.ts`),
-      `export type ${alias} = { __type: '${alias}'; code: object[]; };\n` +
-      `export default { __type: '${alias}', code: JSON.parse(\`${JSON.stringify(json)}\`) } as ${alias};\n`,
+      path.join(outPath, `${fileName.replace('.tz', '.json')}`),
+      JSON.stringify(json),
     );
     $log.info(`successfully compiled ${fileName} to JSON.`);
     return fileName;
   }));
 
-const generateIndex = (compiledFiles: string[]) => {
-  const imports = compiledFiles
+const generateTypes = () => new Promise<string>((resolve, reject) => {
+  child.exec(`yarn generate-types`, (err, stdout, stderr) => {
+    if (stderr) {
+      $log.error(stderr);
+    }
+    err ? reject(err) : resolve(stdout);
+  });
+});
+
+const generateIndex = (codeFiles: string[]) => {
+  const fileNames = codeFiles
+    .map(x => x.replace('.code.ts', ''));
+
+  const codeImports = fileNames
     .map(fileName =>
-      `import ${fileNameToTitleCase(fileName)}, { ${fileNameToAlias(fileName)} } from './${path.relative(process.cwd(), outPath)}/${fileName}';`,
+      `import { ${fileNameToCodeObject(fileName)} } from './${path.relative(process.cwd(), outPath)}/${fileName}.code';\n` +
+      `import { ${fileNameToContractType(fileName)} } from './${path.relative(process.cwd(), outPath)}/${fileName}.types';`
+      ,
     )
     .join('\n');
 
-  const exports = compiledFiles
-    .map(fileName => `  ${fileNameToTitleCase(fileName)},\n  ${fileNameToAlias(fileName)},`)
+  const exports = codeFiles
+    .map(fileName => `  ${fileNameToContractType(fileName)},\n  ${fileNameToCodeObject(fileName)},`)
     .join('\n');
 
-  return `${imports}\n\nexport {\n${exports}\n};\n`;
+  return `${codeImports}\n\nexport {\n${exports}\n};\n`;
 };
 
 (async () => {
   try {
+    fs.promises.mkdir(outPath);
     const [contracts] = await Promise.all([
       // Retrieve compiled Michelson
       fs.promises
         .readdir(michelsonPath)
         .then(files => files.filter(f => path.extname(f) === '.tz')),
-      // Clear any existing .ts files in the output path
+      // Clear any existing .json and .ts files in the output path
+      emptyDirectory(outPath, f => path.extname(f) === '.json'),
       emptyDirectory(outPath, f => path.extname(f) === '.ts'),
     ]);
-    const compiledFiles = await compile(contracts);
+
+    // Compile json
+    await compileToJsonFiles(contracts);
+
+    // Generate Types
+    await generateTypes();
+
+    const outputFiles = await fs.promises.readdir(outPath);
+    const codeFiles = outputFiles
```

Comment:
Do we need to poll the directory again here? I believe the `contracts` variable would suffice here (and would save a `filter` call).

Reply:
This is a different directory than the original